libraries/ms-helper/inc/thermistor_converter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+// Thermistor Reading interface - converts an ADC voltage reading to temperature
+// Requires GPIO and Interrupts to be initialized.
+#include "adc.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+
+#define RESISTANCE_LEN 71
```

Comment:
What is this for? I don't think this belongs in the header.

You also need to fix your naming.

Reply:
This is the length of the array which holds all of the resistance values. I thought it would be better to use a #define rather than a number at its declaration

Reply:
There's no reason to expose that, so it doesn't need to be in the header.

You also don't need to explicitly set the array length. If you declare the variable like `int foo[] = { ... }`, it will automatically size itself to fit the contents provided. You can then use the macro `SIZEOF_ARRAY` to determine the array length at compile-time. Note that this macro only works for arrays, not pointers. There's a subtle difference. Arrays can decay into pointers.

Reply:
How would I use `SIZEOF_ARRAY` to replace the old macro in this context `for (uint16_t i = 0; i < RESISTANCE_LEN; i++) {` ?